### PR TITLE
Add event info tooltips and rankings links to the events tab and the schedule tab

### LINKED ISSUE #44: Add event info tooltips and rankings links to the events tab and the schedule tab
The `EventTitle` component on the events tab and the schedule tab of the contest page should link to the event's rankings (use the `linkToRankings` attribute for that) and include a tooltip with the event description (use the `showDescription` tooltip for that).

### PR DESCRIPTION
**Pull request by zbaruch20**
Resolves #44

### CODE REVIEW

File: client/app/competitions/[id]/events/page.tsx [hunk 1]
```diff
@@ -32,28 +32,35 @@ const ContestEventsPage = async ({ params }: { params: { id: string } }) => {
             {contest.events.map((compEvent) =>
               compEvent.rounds.map((round, roundIndex) => {
                 const cutoffText = round.cutoff
-                  ? `${round.cutoff.numberOfAttempts} ${
-                    round.cutoff.numberOfAttempts === 1 ? 'attempt' : 'attempts'
+                  ? `${round.cutoff.numberOfAttempts} ${round.cutoff.numberOfAttempts === 1 ? 'attempt' : 'attempts'
```
Comment (dmint789) [inline client/app/competitions/[id]/events/page.tsx:35]: Are you using Prettier and is it reading the prettierrc file in the root of the project? This change seems unintentional.

Comment (dmint789) [reply client/app/competitions/[id]/events/page.tsx:35]: Same for some of the other changes in this PR

File: client/app/globals.css [hunk 2]
```diff
@@ -4,6 +4,7 @@
 }
 
 body[data-bs-theme='dark'] {
+
```
Comment (dmint789) [inline client/app/globals.css:7]: What is this and the other newline for?

**Review by dmint789 — CHANGES_REQUESTED** [review]

File: client/app/competitions/[id]/events/page.tsx (hunk 1 above)
Comment (zbaruch20) [reply client/app/competitions/[id]/events/page.tsx:35]: I ran a format via VSCode, and then committing ran ESLint but not sure about Prettier

File: client/app/globals.css (hunk 2 above)
Comment (zbaruch20) [reply client/app/globals.css:7]: Assuming just formatting

File: client/app/competitions/[id]/events/page.tsx (hunk 1 above)
Comment (dmint789) [reply client/app/competitions/[id]/events/page.tsx:35]: Would you mind installing the Prettier vs code plugin, setting it as your default formatter and reformatting this file?

**Comment by zbaruch20** [comment]
I used prettier on the files in question

File: client/app/globals.css [hunk 3]
```diff
@@ -126,3 +126,8 @@ body[data-bs-theme='light'] {
     white-space: nowrap;
   }
 }
+
+/* Override Bootstrap's overflow-x for responsible tables (only in contest layouts) */
+.cc-contest-layout .table-responsive {
+  overflow-x: visible;
```
Comment (dmint789) [inline client/app/globals.css:132]: Why is this necessary? I just tried it out, and it looks like it works fine without this, but with this style added, scrolling is disabled on mobile, which doesn't have enough width to display the entire schedule.

**Review by dmint789 — CHANGES_REQUESTED** [review]

File: client/app/globals.css (hunk 3 above)
Comment (zbaruch20) [reply client/app/globals.css:132]: If an event has a large description, the tooltip box could mess with the responsive table rendering creating a weird scroll bar.

image

Comment (zbaruch20) [reply client/app/globals.css:132]: image

Comment (dmint789) [reply client/app/globals.css:132]: Ohhh, good catch! Shouldn't it be overflow-y: visible in that case though? Like I said, this disables horizontal scrolling on mobile.

Comment (zbaruch20) [reply client/app/globals.css:132]: I experimented with `overflow-y` but it didnt seem to do anything. we might need to redesign the tooltip and/or table design to get around this, or may just need to use the default bootstrap tooltips. Or perhaps make the tooltips size 0 until the button is hovered.

Comment (dmint789) [reply client/app/globals.css:132]: The Bootstrap tooltips require JS, but we just use the Bootstrap CSS classes. I'll try to play around with this a bit.

Comment (dmint789) [reply client/app/globals.css:132]: I'll just look into it separately, I just realized there's no reason to be holding up this PR over this

**Review by dmint789 — APPROVED** [review]
